pkgs/by-name/ea/easytier/package.nix

```diff
@@ -0,0 +1,43 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  rustPlatform,
+  protobuf,
+
+  # with QUIC protocol support
+  withQuic ? false,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "easytier";
+  version = "1.2.3";
+
+  src = fetchFromGitHub {
+    owner = "EasyTier";
+    repo = "EasyTier";
+    rev = "v${version}";
+    hash = "sha256-7T6xdJrVTgg7rSTC2PaVTsBTgi14qJzaR6M8tRUL8OQ=";
+  };
+
+  cargoHash = "sha256-9wAGUVYKz7+Q8y+dmay8pEZnv7PikzuAoas/h5T3sLE=";
+
+  nativeBuildInputs = [ protobuf ];
+
+  buildNoDefaultFeatures = stdenv.isMips;
+  buildFeatures = lib.optional stdenv.isMips [ "mips" ] ++ lib.optional withQuic [ "quic" ];
+
+  doCheck = false; # tests failed due to heavy rely on network
```

Comment:
Can we only disable some of them? See https://nixos.org/manual/nixpkgs/unstable/#disabling-package-tests

Reply:
Thx for your review! Almost all of the tests for this program rely heavily on the network or on the target host's device. E.g., if the target host does not support IPV6, all related tests will fail. I was able to pass the offline check on my host by skipping nearly half of the tests. but I'm not sure if it works on other hosts.